<?xml version="1.0"?>

<!--
This launch file creates a nodelet manager and adds the drivers for both SA cameras to it.
-->

<launch>
  <node pkg="nodelet" type="nodelet" name="sa_camera_nodelet_manager"  args="manager" output = "screen"/>

  <node pkg="nodelet" type="nodelet" name="left_sa_camera_driver"
        args="load pgr_camera/camera_nodelet sa_camera_nodelet_manager" output="screen">
    <param name="serial" value="12502220" />
    <param name="frame_id" type="str" value="l_situational_awareness_camera_optical_frame" />

    <!-- ROI is applied before rotating image in camera hardware -->
    <param name="roi_offset_x" value="190" />
    <param name="roi_offset_y" value="70" />
    <param name="roi_width" value="900" />
    <param name="roi_height" value="884" />

    <remap from="/left_sa_camera_driver/image" to="/sa/left/image_raw" />
    <remap from="camera_info" to="/sa/left/camera_info" />
  </node>

  <node pkg="nodelet" type="nodelet" name="right_sa_camera_driver"
        args="load pgr_camera/camera_nodelet sa_camera_nodelet_manager" output="screen">
    <param name="serial" value="12491354" />
    <param name="frame_id" type="str" value="r_situational_awareness_camera_optical_frame" />

    <!-- ROI is applied before rotating image in camera hardware -->
    <param name="roi_offset_x" value="190" />
    <param name="roi_offset_y" value="70" />
    <param name="roi_width" value="900" />
    <param name="roi_height" value="884" />

    <remap from="/right_sa_camera_driver/image" to="/sa/right/image_raw" />
    <remap from="camera_info" to="/sa/right/camera_info" />
  </node>

</launch>

